src/SurfacePlot/SurfacePlot.tsx

```diff
@@ -0,0 +1,202 @@
+                }
+              },
+              yaxis: {
+                color: theme.palette.mode === "light" ? "" : "white",
+                gridcolor:
+                  theme.palette.mode === "light"
+                    ? ""
+                    : theme.palette.grey["500"],
+                title: {
+                  font: {
+                    family: "Montserrat",
+                    size: 12
+                  },
+                  text: ylabel || ""
+                }
+              },
+              zaxis: {
+                color: theme.palette.mode === "light" ? "" : "white",
+                gridcolor:
+                  theme.palette.mode === "light"
+                    ? ""
+                    : theme.palette.grey["500"],
+                title: {
+                  font: {
+                    family: "Montserrat",
+                    size: 12
+                  },
+                  text: zlabel || ""
+                }
+              }
+            }
+          }}
+          style={{ flexGrow: 1, height: "100%", minHeight: 0, width: "100%" }}
+          useResizeHandler={true}
+          config={config}
+        />
+      </Box>
+    </ConditionalDialog>
+  );
+};
+
+export default SurfacePlot;
+
+// svg path for fullscreen icon in plotly menu bar
+const fullscreenIcon = {
+  height: 1792,
+  path: "M256 1408h1280v-768h-1280v768zm1536-1120v1216q0 66-47 113t-113 47h-1472q-66 0-113-47t-47-113v-1216q0-66 47-113t113-47h1472q66 0 113 47t47 113z",
+  width: 1792
+};
+
+type ConfigProps = {
+  isFullscreen: boolean;
+  handleClickFullscreen: () => void;
+};
+
+/**
+ * Returns a plotly config function with provided callback for clicking fullscreen
+ */
+const getConfig = ({ isFullscreen, handleClickFullscreen }: ConfigProps) => {
+  return {
+    displaylogo: false, // never display plotly logo
+    modeBarButtonsToAdd: !isFullscreen // if we are not in full screen, show a button to launch to fullscreen
+      ? [
+          {
+            click: handleClickFullscreen,
+            direction: "up",
+            icon: fullscreenIcon,
+            name: "Fullscreen",
+            title: "Fullscreen"
+          }
+        ]
+      : []
+  };
+};
+
+type ConditionalDialogProps = {
+  condition: boolean;
+  onClose: () => void;
+  children: React.ReactNode;
+  dialogTitle?: string;
+};
+
+/**
+ * Returns children in a fullscreen dialog if condition is true, otherwise just returns children
+ */
+function ConditionalDialog({
+  condition,
+  onClose,
+  children,
+  dialogTitle
+}: ConditionalDialogProps) {
+  if (condition) {
+    return (
+      <Dialog maxWidth="xl" fullWidth open>
+        <DialogTitle onClose={onClose}>
+          <Box sx={dialogTitle ? {} : { p: 2 }}>{dialogTitle}</Box>
+        </DialogTitle>
+        <DialogContent dividers>
+          <Box style={{ height: "calc(100vh - 168px)" }}>{children}</Box>
+        </DialogContent>
+      </Dialog>
+    );
+  } else {
+    return children;
+  }
+}
```

Comment:
This code looks identical to the code in LinePlot. Probably it's worth to convert this to it's open component, something like FullscreenDialog or such like.